app/sources/HFTemplateController.m

```diff
@@ -526,15 +526,21 @@ - (HFTemplateNode *)currentSection {
 }
 
 - (void)addEntryWithLabel:(NSString *)label value:(NSString *)value length:(unsigned long long *)length offset:(unsigned long long *)offset {
-    HFTemplateNode *node = [[HFTemplateNode alloc] initWithLabel:label value:value];
-    if (length && offset) {
-        node.range = HFRangeMake(self.anchor + *offset, *length);
-    } else if (length) {
-        node.range = HFRangeMake(self.anchor + self.position, *length);
+    HFTemplateNode *currentNode = self.currentNode;
+    HFTemplateNode *newNode = [[HFTemplateNode alloc] initWithLabel:label value:value];
+    if (length) {
+        if (offset) {
+            newNode.range = HFRangeMake(self.anchor + *offset, *length);
+        } else {
+            newNode.range = HFRangeMake(self.anchor + self.position, *length);
+        }
+        unsigned long long newloc = MIN(currentNode.range.location, newNode.range.location);
```

Comment:
In addition to making a new HFRange for the node being added, an additional HFRange is made by calculating the union of the range of the new node and the range of its parent; then the union is assigned as the parent's range property, reflecting the fact that it "contains" whatever range the new node is associated with.